.travis.yml

```diff
@@ -140,21 +138,24 @@ jobs:
                   ./.docker/docker.sh manifest_list_beta "10" "default"
                   ./.docker/docker.sh manifest_list_beta "" "default"
 
+                elif [ "${TRAVIS_BRANCH}" == "dev" ]; then
```

Comment:
Does this mean we can only tag from dev branch ? I thought we wanted to be able to tag anything manually - but it must go to nodered/node-red-dev

Reply:
Looks like you are confused about tags - there are two types of tags: github release tags (starting with a `v`) and the docker images and manifest list tags.

1. If the origin branch is `master` and the github release tag starts with a `v`, then all images and manifest lists are pushed to `nodered/node-red`. The tags of the docker images are derived from the the github tag (without the `v`) and the manifest list tags are derived from the github tag as well plus the tag `latest`. (as we have now)

2. Else if the origin branch is `dev` and the github release tag contains `beta`, then all images and manifest lists are pushed to `nodered/node-red-dev`. The tags of the docker images are derived from the the github tag (without the `v`) and the manifest list tags are derived from the github tag as well plus the tag `beta`. (same as above, but instead `nodered/node-red` all goes to `nodered/node-red-dev` and `latest` becomes `beta`).

3. Else if the origin branch is `dev`, then all images and manifest lists are pushed to `nodered/node-red-dev`. The tags of the docker images are derived from the the github tag (without the `v`) and the manifest list tags are derived from the github tag as well plus the tag `test`. (same as above, but instead `nodered/node-red` all goes to `nodered/node-red-dev` and `beta` becomes `test`).

4. Else we do nothing - this case will not occur, because we only have two branches which are allowed as origin to publish to docker hub AND a github tag should exist and start with a `v`.

Replacing:
 
```shell
elif [ "${TRAVIS_BRANCH}" == "dev" ];
```

with: 

```shell
elif [ "${TRAVIS_BRANCH}" != "master" ];
```
makes it possible to publish from **any** branch  other then `master` when a github tag (starting with a `v`) has been given. Thought we agreed that we didn't want that.

`any` branch -> PR -> merge into `dev` branch ->  `nodered/node-red-dev` -> PR -> merge into `master` -> `nodered/node-red`

Hope this clarify things.

Reply:
no - we wanted to be able to tag from any branch. So (for example) you can have a branch and try things and so can I  - before they get merged to dev.